python/ENTERPRISE_SUPPORT_PLAN_ENABLED/ENTERPRISE_SUPPORT_PLAN_ENABLED.py

```diff
@@ -0,0 +1,400 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  ENTERPRISE_SUPPORT_PLAN_ENABLED
+Description:
+  Check whether the Enterprise Support Plan is enabled for an AWS Account.
+Trigger:
+  Periodic
+Reports on:
+  AWS::::Account
+Rule Parameters:
+  None
+Scenarios:
+  Scenario: 1
+     Given: AWS Support API call errors out with SubscriptionRequired exception while using the DescribeSeverityLevels API call
+      Then: Return NON_COMPLIANT with Annotation "The AWS Enterprise Support Plan is not enabled for this AWS Account."
+  Scenario: 2
+     Given: The AWS Account does not have access to "critical" case severity level as returned by the AWS Support API DescribeSeverityLevels
+      Then: Return NON_COMPLIANT with Annotation "The AWS Enterprise Support Plan is not enabled for this AWS Account."
+  Scenario: 3
+     Given: The AWS Account has access to "critical" case severity level as returned by the AWS Support API DescribeSeverityLevels
+      Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+from botocore.exceptions import ClientError
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+# AWS Support API has only one endpoint which is in us-east-1 region
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    support_client = get_client('support', event, 'us-east-1')
+    try:
+        # Account with Basic or Developer support returns ClientError exception
+        response = support_client.describe_severity_levels()
+    except ClientError as error:
+        if error.response['Error']['Code'] == 'SubscriptionRequiredException':
+            return build_evaluation(
+                event['accountId'],
+                "NON_COMPLIANT",
+                event,
+                DEFAULT_RESOURCE_TYPE,
+                annotation="The AWS Enterprise Support Plan is not enabled for this AWS Account."
+            )
+        print(error.response['Error']['Code'])
+        raise
+
+    for level in response['severityLevels']:
+        if level['code'] == 'critical':
+            return build_evaluation(event['accountId'], "COMPLIANT", event)
+
+    return build_evaluation(
+        event['accountId'],
+        "NON_COMPLIANT",
+        event,
+        DEFAULT_RESOURCE_TYPE,
```

Comment:
No need to write DEFAULT_RESOURCE_TYPE here. It is passed by default by the function.